features/fields/index.ts

```diff
@@ -176,15 +178,26 @@ export const getColumnNameLabel = (...args: any[]) => {
 /* Returns the filtered column based on their visibility settings. */
 export const getFilteredColumns = (
   columns: Column[],
-  view: Views
+  view?: string
 ): Column[] => {
   if (isArray(columns)) {
     return (
       columns
         // Remove fields that should be hidden on index
-        .filter((column: Column) =>
-          column.baseOptions.visibility.includes(view)
-        )
+        .filter((column: Column) => {
+          switch (view) {
+            case "index":
+              return column.baseOptions.visibleOnIndex;
+            case "show":
+              return column.baseOptions.visibleOnShow;
+            case "edit":
+              return column.baseOptions.visibleOnEdit;
+            case "new":
+              return column.baseOptions.visibleOnNew;
+            default:
+              return true;
```

Comment:
Shouldn't the fallback be `false`?

Reply:
It is true bcs the filtering shouldn't be enforced when view is not passed or different from allowed ones.